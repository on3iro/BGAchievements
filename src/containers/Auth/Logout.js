import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Redirect } from 'react-router-dom'

import * as selectors from './ducks/selectors'
import { logout } from './ducks/actions'

export class Logout extends Component {
  componentWillMount () {
    if (this.props.isLoggedIn) this.props.logout()
  }

  render () {
    return !this.props.isLoggedIn && (
      <Redirect to={{
        pathname: '/login'
      }} />
    )
  }
}

Logout.propTypes = {
  isLoggedIn: PropTypes.bool,
  logout: PropTypes.func.isRequired
}

export const mapStateToProps = (state) => {
  return {
    isLoggedIn: selectors.getLoggedIn(state)
  }
}

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    logout
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Logout)
