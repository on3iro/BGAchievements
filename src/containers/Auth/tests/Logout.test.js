import React from 'react'
import { shallow } from 'enzyme'
import { Redirect } from 'react-router-dom'

import { Logout, mapStateToProps, mapDispatchToProps } from '../Logout'

const renderComponent = newProps => {
  const props = {
    isLoggedIn: false,
    logout: jest.fn(),
    ...newProps
  }

  const renderedWrapper = shallow(<Logout {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<Logout />', () => {
  it('should render <Redirect /> if user is not logged in', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Redirect).length).toMatchSnapshot()
  })

  it('should return null if user is logged in', () => {
    const { renderedWrapper } = renderComponent({isLoggedIn: true})
    expect(renderedWrapper.type()).toMatchSnapshot()
  })

  it('should call logout() if user is logged in', () => {
    const { renderedWrapper, props } = renderComponent({isLoggedIn: true})
    expect(props.logout).toHaveBeenCalled()
  })

  describe('react-redux', () => {
    it('should map state to props', () => {
      expect(mapStateToProps({
        auth: { user: { isLoggedIn: true }}
      })).toMatchSnapshot()
    })
  })

  it('should map dispatch to props', () => {
    expect(mapDispatchToProps(() => null)).toMatchSnapshot()
  })
})
