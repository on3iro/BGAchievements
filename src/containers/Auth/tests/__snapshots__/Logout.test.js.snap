exports[`<Logout /> react-redux should map state to props 1`] = `
Object {
  "isLoggedIn": true,
}
`

exports[`<Logout /> should map dispatch to props 1`] = `
Object {
  "logout": [Function],
}
`

exports[`<Logout /> should render <Redirect /> if user is not logged in 1`] = `1`

exports[`<Logout /> should return null if user is logged in 1`] = `null`
